lib/Parse/ParseDecl.cpp

```diff
@@ -2986,9 +2986,26 @@ ParserStatus Parser::parseNewDeclAttribute(DeclAttributes &Attributes,
     if (!Tok.is(tok::l_paren)) {
       // Normal access control attribute.
       AttrRange = Loc;
-      DuplicateAttribute = Attributes.getAttribute<AccessControlAttr>();
-      if (!DuplicateAttribute)
+      
+      // Try to find a previously parsed access control attribute.
+      auto PrevAccessControlAttr = Attributes.getAttribute<AccessControlAttr>();
+      
+      if (!PrevAccessControlAttr) {
         Attributes.add(new (Context) AccessControlAttr(AtLoc, Loc, access));
+      } else {
```

Comment:
The old duplication error is still preferable when they are actually duplicates, i.e. the parsed access levels match.

Reply:
Done!